api.js

```diff
@@ -1040,9 +1078,15 @@ module.exports = async function attachAPI(app, {wss, db}) {
         return
       }
 
+      const serializedUser = await serialize.user(user)
+
+      // We will also provide the stored email address, but only
+      // because this IS the user and it is therefore safe to give it away.
+      Object.assign(serializedUser, { email: user.email || null })
+
```

Comment:
I'm not feeling great about attaching data outside of `serialize` - perhaps `serialize.user` could take a sessionID property? Compare with how we attach unread messages count to channels when a session ID is passed:

```js
channelBrief: async (c, sessionUser = null) => {
  const obj = {
    id: c._id,
    name: c.name
  }

  if (sessionUser) {
    obj.unreadMessageCount = await getUnreadMessageCountInChannel(sessionUser, c._id)
  }

  return obj
}
```

So we'd do basically the same with users:

```js

user: async (u, sessionUser = null) => {
  const obj = {
    id: u._id,
    username: u.username,
    avatarURL: emailToAvatarURL(u.email || u._id),
    permissionLevel: u.permissionLevel,
    online: await isUserOnline(u._id)
  }
  
  if (sessionUser && sessionUser._id === u._id) {
    obj.email = u.email || null
  }

  return obj
}
```

(That means doing `serialize.user(user, user)`, which admittedly looks a little silly, but *is* consistent with `serialize.channel`! :shipit:)